download_ssara_rsmas.py

```diff
@@ -72,35 +72,39 @@ def generate_ssaraopt_string(template_file):
     """
     # use ssaraopt.platform, relativeOrbit and frame if given, else use ssaraopt
     template_options = Template(template_file).get_options()
-    try:
-       platform = template_options['ssaraopt.platform']
-       relativeOrbit = template_options['ssaraopt.relativeOrbit']
-       frame = template_options['ssaraopt.frame']
-       ssaraopt='--platform='+platform+' --relativeOrbit='+relativeOrbit+' --frame='+frame
-
-       try:
-          startDate = template_options['ssaraopt.startDate']
-          ssaraopt=ssaraopt+' -s='+startDate
-       except:
-          pass
-       try:
-          endDate = template_options['ssaraopt.endDate']
-          ssaraopt=ssaraopt+' -e='+endDate
-       except:
-          pass
-
-    except:
-       try: 
-         ssaraopt = template_options['ssaraopt']
-       except:
-         raise Exception('no ssaraopt or ssaraopt.platform, relativeOrbit, frame found')
-
-    # add parallel doenload option. If ssaraopt.parallelDownload not given use default value
-    try:
-       parallelDownload = template_options['ssaraopt.parallelDownload']
-    except:
-       parallelDownload = '30'     # default
-    ssaraopt=ssaraopt+' --parallel='+parallelDownload
+
+    ssaraopt_keys = ['ssaraopt.platform', 'ssaraopt.relativeOrbit', 'ssaraopt.frame', 'ssaraopt']
+
+    bad_key=False
```

Comment:
How about this?
```python
ssaraopt_keys = ['ssaraopt.platform', 'ssaraopt.relativeOrbit', 'ssaraopt.frame']

bad_key = False
for key in ssaraopt_keys:
    if key not in template_options:
        bad_key = True
        if 'ssaraopt' not in template_options:
            raise Exception('no ssaraopt or ssaraopt.platform, relativeOrbit, frame found')
        else:
            ssaraopt = template_options['ssaraopt']
            break

if not bad_key:
    platform = template_options['ssaraopt.platform']
    relativeOrbit = template_options['ssaraopt.relativeOrbit']
    frame = template_options['ssaraopt.frame']
    ssaraopt='--platform={} --relativeOrbit={} --frame={}'.format(platform, relativeOrbit, frame)
    ...
```

Reply:
Seems a bit cleaner to deal with the `ssaraopt` key separately

Reply:
Your right, I was trying to come up with a not bad way to do that, but missed this easy fix.